src/briefcase/platforms/macOS/__init__.py

```diff
@@ -577,48 +574,46 @@ def package_app(
         """Package an app bundle.
 
         :param app: The application to package
-        :param sign_app: Should the application be signed? Default: ``True``
         :param notarize_app: Should the app be notarized? Default: ``True`` if the
             app has been signed with a real identity; ``False`` if the app is
             unsigned, or an ad-hoc signing identity has been used.
         :param identity: The code signing identity to use. This can be either
             the 40-digit hex checksum, or the string name of the identity.
             If unspecified, the user will be prompted for a code signing
             identity. Ignored if ``sign_app`` is ``False``.
-        :param adhoc_sign: If ``True``, code will be signed with adhoc identity of "-"
+        :param adhoc_sign: If ``True``, code will be signed with ad-hoc identity
+            of "-", and the resulting app will not be re-distributable.
         """
-        if sign_app:
-            if adhoc_sign:
-                if notarize_app:
-                    raise BriefcaseCommandError(
-                        "Can't notarize an app with an adhoc signing identity"
-                    )
-
-                identity = "-"
-                self.logger.info(
-                    "Signing app with adhoc identity...", prefix=app.app_name
+        if adhoc_sign:
+            if notarize_app:
+                raise BriefcaseCommandError(
+                    "Can't notarize an app with an ad-hoc signing identity"
                 )
-            else:
-                # If we're signing, and notarization isn't explicitly disabled,
-                # notarize by default.
-                if notarize_app is None:
-                    notarize_app = True
 
-                identity, identity_name = self.select_identity(identity=identity)
+            identity = "-"
+            self.logger.info(
+                (
+                    "Signing app with ad-hoc identity... Note that this app "
+                    "will run, but cannot be re-distributed."
+                ),
```

Comment:
This should be 2 messages - an info saying that the ad-hoc identiy being used and a warning that the app won't be redistributable.